pymc_experimental/tests/distributions/test_discrete.py

```diff
@@ -118,3 +121,30 @@ def test_moment(self, mu, lam, size, expected):
         with pm.Model() as model:
             GeneralizedPoisson("x", mu=mu, lam=lam, size=size)
         assert_moment_is_expected(model, expected)
+
+
+class TestBetaNegativeBinomialClass:
+    """
+    Wrapper class so that tests of experimental additions can be dropped into
+    PyMC directly on adoption.
+    """
+
+    def test_logp(self):
+        def factln(n):
+            return gammaln(n + 1)
+
+        def logp(value, alpha, beta, r):
+            return (
+                betaln(r + value, alpha + beta)
+                - betaln(r, alpha)
+                + gammaln(value + beta)
+                - factln(value)
+                - gammaln(beta)
+            )
```

Comment:
I thought there would be a scipy implementation we could check against. This is suboptimal, because we are just duplicating the logp code in two different formats and checking they match. If there is no scipy implementation, I suggest picking a couple of values from another implementation such as:
* https://search.r-project.org/CRAN/refmans/brr/html/BNBDist.html
* https://rdrr.io/cran/extraDistr/man/BetaNegBinom.html

And testing we get the same here. An example of this approach is done for the ExGaussian: https://github.com/pymc-devs/pymc/blob/ec4407d3db2c8ebe77e62c65f2eda3f5703e6279/tests/distributions/test_continuous.py#L994-L1021

We shoudn't write the test like that, it's too inefficient, but something like:
```python
def test_logp(self):
  alpha, beta, r, value = pt.scalars("alpha", "beta", "r", "value")
  logp = pm.logp(BetaNegativeBinomial.dist(alpha, beta, r), value)
  logp_fn = pytensor.function([value, alpha, beta, r], logp)

  for test_value, test_alpha, test_beta, test_r in (...):
    np.testing.assert_allclose(logp_fn(test_value, test_alpha, test_beta, test_r), expected_logp)
```

Reply:
Makes sense

Reply:
Implemented. Bit far from my R days, but put the code there for reference. Let me know if the tested values match your expectation